pkg/webhook/validating/network.go

```diff
@@ -275,3 +286,30 @@ func checkNetworkTypeExist(ctx context.Context, client client.Reader, networkTyp
 	}
 	return false, "", nil
 }
+
+func checkUnderlayNetworkOverlapped(ctx context.Context, c client.Reader, nodeSelector map[string]string) (bool, string, error) {
+	networks := &networkingv1.NetworkList{}
+	if err := c.List(ctx, networks); err != nil {
+		return false, "", err
+	}
+
+	nodeList, err := controllerutils.ListActiveNodesToNames(ctx, c, client.MatchingLabels(nodeSelector))
+	if err != nil {
+		return false, "", fmt.Errorf("unable to update node list %v", err)
+	}
+
+	nodeListMap := utils.StringSliceToMap(nodeList)
+
+	for i := range networks.Items {
+		if networkingv1.GetNetworkType(&networks.Items[i]) != networkingv1.NetworkTypeUnderlay {
```

Comment:
should the network which is being checked be skipped here?

Reply:
fixed